### PR TITLE
Test on Django 4.2

### PR DESCRIPTION
**Pull request by adamchainz**
Missing psycopg3 support, as previously discussed that is a more involved change.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -81,7 +82,7 @@ packaging = ">=19.2"
 Pillow = "9.2.0"
 poetry-core = "1.3.2"
 pre-commit = "2.13.0"
-psycopg2-binary = "2.8.6"
+psycopg2-binary = "2.9.6"
```
Comment (adamchainz) [inline pyproject.toml:85]: Upgraded only to get the M1 macOS wheel for easy local install, it's not that Django's psycopg2 support requires a newer version ([docs](https://docs.djangoproject.com/en/4.2/ref/databases/#postgresql-notes)).

File: tox.ini [hunk 2]
```diff
@@ -1,13 +1,14 @@
 [tox]
 isolated_build = true
-envlist = py37-django22,py37-django32,{38,39,310}-django{22,32,40,41},report
+envlist = py37-django22,py37-django32,py{38,39,310}-django{22,32,40,41,42},report
```
Comment (adamchainz) [inline tox.ini:3]: I noticed the "py" was missing - this might need propagating back to the project template?

**Comment by adamchainz** [comment]
Circle CI ain't running:

> Could not find a usable config.yml, you may have revoked the CircleCI OAuth app.
> Please sign out of CircleCI and log back in with your VCS before triggering a new pipeline.

I did run the tests locally though and noted no new warnings.

**Comment by wesleykendall** [comment]
@adamchainz thanks for the help! I did this upgrade at https://github.com/Opus10/django-pghistory/pull/95 and put you as a `Co-authored-by`. I implemented some of your changes (tox, psycopg2-binary for Macs, etc).

I also made a similar ticket as pgtrigger to allow pghistory SQL injection mechanisms to be configurable - https://github.com/Opus10/django-pghistory/issues/96

These changes were deployed in v2.8.0

**Comment by adamchainz** [comment]
🙏 Thanks!

**Pull request closed without merging**